### ISSUE TITLE
Cannot deserialize BuiltList with only 1 element

### ISSUE THREAD

**Issue by skybur** [google/built_value.dart#594]
When I tried to deserialize `BuiltList` with only 1 element, I got this error:
```
type '_InternalLinkedHashMap<String, Object>' is not a subtype of type 'Iterable<dynamic>' in type cast
```
If the list is empty or contains more than 1 element, this doesn't happen
This is serialized BuiltList with 1 element:
```
{$: list, : {$: String, : A}}
```
and this is with 2 element:
```
{$: list, : [{$: String, : A}, {$: String, : B}]}
```
and this is empty list:
```
{$: list, : []}
```
Is this is intended?

**Comment by skybur** [comment]
It looks like because I also used `StandardJsonPlugin`. Is there any fix or workaround for this?

**Comment by davidmorgan** [comment]
Thanks for the report--this is indeed a bug.

Most users of `StandardJsonPlugin` always specify the types, rather than having them end up in the JSON; otherwise we'd've seen this issue already!

The fix turns out to be straightforward, PR: #595
